clang/include/clang/Tooling/Refactoring/Rename/RenamingAction.h

```diff
@@ -116,6 +117,26 @@ class QualifiedRenamingAction {
   std::map<std::string, tooling::Replacements> &FileToReplaces;
 };
 
+enum class ObjCSymbolSelectorKind {
+  /// The rename location is an Objective-C method call, eg. `[self add: 1]`.
+  MessageSend,
+  
+  /// The rename location is an Objective-C method definition, eg.
+  /// ` - (void)add:(int)theValue`
+  MethodDecl,
+
+  /// It is unknown if the renamed location is a method call or declaration.
+  ///
+  /// The selector kind is being used to improve error recovery, passing unknown
+  /// does not lead to correctness issues.
+  Unknown
+};
+
+llvm::Expected<SmallVector<SourceLocation>> findObjCSymbolSelectorPieces(
```

Comment:
I suspect you might get some push back on this, with the canonical declaration being returning an `Error` and taking a `SmallVector&`.